python/lsst/ap/association/mpSkyEphemerisQuery.py

```diff
@@ -0,0 +1,205 @@
+# This file is part of ap_association.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Solar System Object Query to MPSky in place of a internal Rubin solar
+system object caching/retrieval code.
+
+Will compute the location for of known SSObjects within a visit
+"""
+
+__all__ = ["MPSkyEphemerisQueryConfig", "MPSkyEphemerisQueryTask"]
+
+
+import pandas as pd
+import pyarrow as pa
+import requests
+import os
+
+import lsst.pex.config as pexConfig
+from lsst.utils.timer import timeMethod
+from lsst.ap.association.utils import getMidpointFromTimespan
+from lsst.geom import SpherePoint
+
+from lsst.pipe.base import PipelineTask, PipelineTaskConfig, \
+    PipelineTaskConnections, connectionTypes, Struct, NoWorkFound
+
+
+class MPSkyEphemerisQueryConnections(PipelineTaskConnections,
+                                     dimensions=("instrument",
+                                                 "group", "detector")):
+
+    predictedRegionTime = connectionTypes.Input(
+        doc="The predicted exposure region and time",
+        name="regionTimeInfo",
+        storageClass="RegionTimeInfo",
+        dimensions={"instrument", "group", "detector"},
+    )
+
+    ssObjects = connectionTypes.Output(
+        doc="MPSky-provided Solar System objects observable in this detector-visit",
+        name="preloaded_SsObjects",
+        storageClass="DataFrame",
+        dimensions=("instrument", "group", "detector"),
+    )
+
+
+class MPSkyEphemerisQueryConfig(
+        PipelineTaskConfig,
+        pipelineConnections=MPSkyEphemerisQueryConnections):
+    observerCode = pexConfig.Field(
+        dtype=str,
+        doc="IAU Minor Planet Center observer code for queries "
+            "(Rubin Obs./LSST default is X05)",
+        default='X05'
+    )
+    queryBufferRadiusDegrees = pexConfig.Field(
+        dtype=float,
+        doc="Buffer radius added to camera bounding circle for Ephemeris "
+            "cone search. Defaults to 10 deg/day * 30 minutes",
+        default=0.208
```

Comment:
Yes. It might be overkill, but I don't know of a safe upper bound for the difference between a predictedRegionTime and the actual observation it produces. If this causes performance issues I'm happy to reduce it.

Reply:
Fine, let's wait and see. Grabbing a roughly raft-size area is still a big improvement over the whole focal plane.